File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/adapter/ListAdapterUtils.java
package com.github.rahmnathan.localmovies.app.adapter;

import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.util.Base64;
import android.view.Gravity;
import android.widget.ImageView;
import android.widget.TextView;

import com.github.rahmnathan.localmovies.app.enums.MovieOrder;
import com.github.rahmnathan.localmovies.info.provider.data.Movie;

import java.util.Comparator;
import java.util.List;

import rahmnathan.localmovies.R;

class ListAdapterUtils {

    void sort(List<Movie> movies, MovieOrder order) {
        switch (order) {
            case DATE_ADDED:
                movies.sort((movie1, movie2) -> movie2.getCreated().compareTo(movie1.getCreated()));
                break;
            case MOST_VIEWS:
                movies.sort((movie1, movie2) -> Integer.compare(movie2.getViews(), movie1.getViews()));
                break;
            case RATING:
                movies.sort((movie1, movie2) -> String.valueOf(movie2.getIMDBRating()).compareTo(String.valueOf(movie1.getIMDBRating())));
                break;
            case RELEASE_YEAR:
                movies.sort((movie1, movie2) -> movie2.getReleaseYear().compareTo(movie1.getReleaseYear()));
                break;
            case TITLE:
                movies.sort(Comparator.comparing(Movie::getTitle));
                break;
        }
    }

    void mapTitle(String title, TextView titleView){
        titleView.setText(title);
        titleView.setTextSize(17);
        titleView.setGravity(Gravity.CENTER);
        titleView.setTextColor(Color.WHITE);
    }

    void mapRatings(String imdbRating, String metaRating, TextView ratings){
        ratings.setGravity(Gravity.CENTER);
        ratings.setTextColor(Color.WHITE);
        ratings.setTextSize(12);
        ratings.setText(String.format("IMDB: %s Meta: %s", imdbRating, metaRating));
    }

    void mapImage(String base64Image, ImageView imageView){
        if (base64Image != null && !base64Image.equals("") && !base64Image.equals("null")) {
            byte[] image = Base64.decode(base64Image, Base64.DEFAULT);
            imageView.setImageBitmap(BitmapFactory.decodeByteArray(image, 0, image.length));
        } else {
            imageView.setImageResource(R.mipmap.no_poster);
        }
    }

    void mapYear(String releaseYear, TextView year){
        year.setText(String.format("Release Year: %s", releaseYear));
        year.setTextColor(Color.WHITE);
        year.setGravity(Gravity.CENTER);
        year.setTextSize(12);
    }

    void display(List<Movie> movies, List<Movie> newMovies){
        movies.clear();
        movies.addAll(newMovies);
    }
}

File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/adapter/list/ListAdapterUtils.kt
package com.github.rahmnathan.localmovies.app.adapter.list

import android.graphics.BitmapFactory
import android.graphics.Color
import android.util.Base64
import android.view.Gravity
import android.widget.ImageView
import android.widget.TextView
import com.github.rahmnathan.localmovies.app.data.Media
import com.github.rahmnathan.localmovies.app.data.MovieOrder
import rahmnathan.localmovies.R
import java.util.*

object ListAdapterUtils {

    @JvmStatic
    fun sort(mediaList: List<Media>, order: MovieOrder?) {
        when (order) {
            MovieOrder.DATE_ADDED -> mediaList.sortedByDescending { media -> media.created }
            MovieOrder.RATING -> mediaList.sortedByDescending { media -> media.imdbRating }
            MovieOrder.RELEASE_YEAR -> mediaList.sortedByDescending { media -> media.releaseYear }
            MovieOrder.TITLE -> mediaList.sortedBy { media -> media.title }
        }
    }

    @JvmStatic
    fun mapTitleToView(title: String?, titleView: TextView, fontSize: Int) {
        titleView.text = title
        titleView.textSize = fontSize.toFloat()
        titleView.gravity = Gravity.CENTER
        titleView.setTextColor(Color.WHITE)
    }

    @JvmStatic
    fun mapRatingsToView(imdbRating: String?, metaRating: String?, ratings: TextView) {
        ratings.gravity = Gravity.CENTER
        ratings.setTextColor(Color.WHITE)
        ratings.textSize = 12f
        ratings.text = String.format("IMDB: %s Meta: %s", imdbRating, metaRating)
    }

    @JvmStatic
    fun mapImageToView(base64Image: String?, imageView: ImageView) {
        if (base64Image != null && base64Image != "" && base64Image != "null") {
            val image = Base64.decode(base64Image, Base64.DEFAULT)
            imageView.setImageBitmap(BitmapFactory.decodeByteArray(image, 0, image.size))
        } else {
            imageView.setImageResource(R.mipmap.no_poster)
        }
    }

    @JvmStatic
    fun mapYearToView(releaseYear: String?, year: TextView, fontSize: Int) {
        year.text = String.format("Release Year: %s", releaseYear)
        year.setTextColor(Color.WHITE)
        year.gravity = Gravity.CENTER
        year.textSize = fontSize.toFloat()
    }
}